Documentation/storage.md

```diff
@@ -0,0 +1,161 @@
+# Storage options
+
+Dex requires persisting state to perform various tasks such as track refresh tokens, preventing replays, and rotating keys. This document is a summary of the storage configurations supported by dex.
+
+Regardless of the implementation, dex saves sensitive data in its backing storage: signing keys, bcrypt'd passwords, etc. Storage breaches are serious because attacks may propagate to any application which relies on dex. Whatever storage option used, transport security and database ACLs are both musts.
```

Comment:
I would reword this to be:

**Note:** Storage breaches are serious as they can affect applications that rely on dex. Dex saves a lot of sensitive data in its backing storage, including **signing keys** and **bcrypt'd passwords**. As such, transport security and database ACLs should both be used, no matter which storage option is chosen. (LINK TO SETUP DOCS FOR THIS)